### PR TITLE
create chip group (#395)

### PR DESCRIPTION
**Pull request by PedroMiguel7**
## Description
Creation of chip group with the same behavior as the Chip component and some new ones like selecting multiple chips.

## Screenshots

- multiples:
image

- with dropdown:

https://user-images.githubusercontent.com/106749260/229111359-8d71c288-38c6-435a-ac4e-2414142ec078.mp4

## Storybook
https://62eab350a45bdb0a5818520e-cxiglhwuec.chromatic.com/

## Compliance
 - [x] I have verified that this change complies with our code and contribution policies.
- [x] I have verified that this change does not cause regressions and does not affect other parts of the code.

### CODE REVIEW

File: projects/ion/src/lib/chip-group/chip-group.component.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+  private clearChips(): void {
+    this.chips.forEach((chip) => {
+      chip.selected = false;
+    });
+  }
+
+  private clearChipsWithDropdown(): void {
+    this.chips.forEach((chip) => {
+      chip.selected && chip.options ? (chip.selected = false) : '';
```
Comment (iurynogueira) [inline projects/ion/src/lib/chip-group/chip-group.component.ts:62]:
```suggestion
      chip.selected && chip.options && (chip.selected = false)';
```

check this.

**Review by iurynogueira — CHANGES_REQUESTED** [review]

**Comment by iurynogueira** [comment]
Check unit tests, i send msg in chat.

File: projects/ion/src/lib/chip-group/chip-group.component.ts (hunk 1 above)
Comment (PedroMiguel7) [reply projects/ion/src/lib/chip-group/chip-group.component.ts:62]: Line 62 serves to clean only the chips that have a dropdown, so the ternary checks if the selected chip has options(dropdown), if it does, it is deselected. If not, nothing happens.

File: projects/ion/src/lib/chip-group/chip-group.component.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
+import { Component, EventEmitter, Input, Output } from '@angular/core';
+import { BadgeType } from '../core/types/badge';
+import { ChipSize, IconDirection, IonChipProps } from '../core/types';
+
+export interface ChipInGroup extends IonChipProps {
+  selected?: boolean;
+}
+
+export interface ChipGroupProps {
```
Comment (Deeved) [inline projects/ion/src/lib/chip-group/chip-group.component.ts:9]: move `ChipGroupProps` to core/types

File: projects/ion/src/lib/chip-group/chip-group.component.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  chips?: Array<IonChipProps>;
+  disabled?: boolean;
+  size?: ChipSize;
+  multiple?: boolean;
+  events?: EventEmitter<ChipEvent>;
+  selected: EventEmitter<ChipInGroup>;
```
Comment (Deeved) [inline projects/ion/src/lib/chip-group/chip-group.component.ts:15]:
```suggestion
  selected?: EventEmitter<ChipInGroup>;
```

File: projects/ion/src/lib/chip-group/chip-group.component.ts [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+  selectChip(chipSelected: ChipInGroup): void {
+    if (chipSelected.options) {
+      this.clearChipsWithDropdown();
+    }
+    if (!this.multiple) {
+      this.clearChips();
+    }
+    chipSelected.selected = true;
+    this.selected.emit(chipSelected);
```
Comment (Deeved) [inline projects/ion/src/lib/chip-group/chip-group.component.ts:51]: when clicking on the disabled chip an event is emitted

File: projects/ion/src/lib/chip-group/chip-group.component.ts (hunk 2 above)
Comment (larissacard) [reply projects/ion/src/lib/chip-group/chip-group.component.ts]: done

File: projects/ion/src/lib/chip-group/chip-group.component.scss [hunk 5]
```diff
@@ -0,0 +1,4 @@
+.chip-group {
+  display: flex;
+  gap: 6px;
```
Comment (AlyssonMascarenhas) [inline projects/ion/src/lib/chip-group/chip-group.component.scss:3]: In figma it asks for a gap of 8px, the padding of the items which is 6 12
image

Comment (PedroMiguel7) [reply projects/ion/src/lib/chip-group/chip-group.component.scss:3]: this 6px gap refers to the distance between the chips, it does not change any style of the chip😊

**Review by iurynogueira — APPROVED** [review]

**Review by mizek1 — CHANGES_REQUESTED** [review]
I found 2 problems:

[Screencast from 14-04-2023 13:44:13.webm](https://user-images.githubusercontent.com/57760325/232106140-1dfc5214-8eac-420d-a027-3e7a9f429862.webm)
In this one, selected property is always emitted as true, even if chip isn't selected anymore. It would be easier to manage if it always reflected the selected property.

[Screencast from 14-04-2023 13:43:18.webm](https://user-images.githubusercontent.com/57760325/232106714-d8b898dd-61c3-481f-a801-19d01b8ad8ca.webm)
This one I don't know why is happening, but it is happening. LOL

**Review by AlyssonMascarenhas — APPROVED** [review]

File: projects/ion/src/lib/chip-group/chip-group.component.html [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+    [multiple]="chip.multiple"
+    [infoBadge]="chip.infoBadge || infoBadge"
+    [iconPosition]="chip.iconPosition || iconPosition"
+    [rightBadge]="chip.rightBadge || rightBadge"
+    (click)="selectChip(chip)"
+    (dropdownEvents)="dropdownEvents($event)"
+  ></ion-chip>
+</div>
+
+<!--     [showDropdown]="
```
Comment (Deeved) [inline projects/ion/src/lib/chip-group/chip-group.component.html:20]: remove commented code

**Review by Deeved — APPROVED** [review]

**Pull request merged**